gallery/src/PageHeader/HostedVersionsSelect.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, ChangeEventHandler, useState } from 'react';
+import { NxFormSelect } from '@sonatype/react-shared-components';
+
+import packageJson from '../../package.json';
+
+import './HostedVersionsSelect.scss';
+
+const CURRENT_VERSION = packageJson.version;
+const HOSTED_VERSIONS_JSON_URL = 'https://gallery.sonatype.dev/hosted-versions.json';
+
+const getHostedVersions = () => fetch(HOSTED_VERSIONS_JSON_URL).then((response) => {
```

Comment:
If this function had an explicit return type you probably wouldn't need as many type annotations in `useEffect`

Reply:
> If this function had an explicit return type you probably wouldn't need as many type annotations in `useEffect`

I think this is cleaned up here:
457e155bea405b56213e7592b3be25760726364b
18aab428bcb0badf4d66b08b695127dd78b9a43e